build_library/modify_image_util.sh

```diff
@@ -83,7 +83,13 @@ finish_modify_image() {
     cleanup_mounts "${ROOT_FS_DIR}"
     trap - EXIT
 
-    upload_image "${DST_IMAGE}"
+    declare -a image_files
+    declare -a compressed_images
+
+    image_files+=( "${DST_IMAGE}" )
+    compress_disk_images image_files compressed_images
+
+    upload_image "${compressed_images[@]}"
```

Comment:
Same here, the image may not get compressed if the extension mismatches

Reply:
Is there any situation in which this would not be a compress-able image (with a matching extension)?